lib/net.c

```diff
@@ -219,6 +233,120 @@ ipv4_ntuple_filter_add(struct gatekeeper_if *iface, rte_be32_t dst_ip_be,
 	return 0;
 }
 
+static int
+ipv4_flow_add(struct gatekeeper_if *iface, rte_be32_t dst_ip_be,
+	rte_be16_t src_port_be, rte_be16_t src_port_mask_be,
+	rte_be16_t dst_port_be, rte_be16_t dst_port_mask_be,
+	uint8_t proto, uint16_t queue_id)
+{
+	struct rte_flow_attr attr = { .ingress = 1, };
+	struct rte_flow_action_queue queue = { .index = queue_id };
+	struct rte_flow_action action[2] = {
+		{
+			.type = RTE_FLOW_ACTION_TYPE_QUEUE,
+			.conf = &queue,
+	       	},
+		{
+			.type = RTE_FLOW_ACTION_TYPE_END,
+		}
+	};
+	struct rte_flow_item pattern[3] = {
```

Comment:
Not only is the number 3 insufficient (see another comment), but we can drop it since the compiler will calculate the correct number for us.